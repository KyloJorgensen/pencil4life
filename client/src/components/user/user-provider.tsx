'use strict';

import * as React from 'react';

export interface IUserContext extends UserWrapperMethods, IUserState{
}

const Context: IUserContext = {
  signup: () => {
    console.error('Error: Cannot signup no UserContext.Provider element in parents');
  },
  getUser: () => {
    console.error('Error: Cannot getUser no UserContext.Provider element in parents');
  },
  login: () => {
    console.error('Error: Cannot login no UserContext.Provider element in parents');
  },
  logout: () => {
    console.error('Error: Cannot logout no UserContext.Provider element in parents');
  },
  updateUser: () => {
    console.error('Error: Cannot updateUser no UserContext.Provider element in parents');
  },
  updatePassword: () => {
    console.error('Error: Cannot updatePassword no UserContext.Provider element in parents');
  },
  forgotPassword: () => {
    console.error('Error: Cannot forgotPassword no UserContext.Provider element in parents');
  },
  resetPassword: () => {
    console.error('Error: Cannot resetPassword no UserContext.Provider element in parents');
  },
  checkResetCode: () => {
    console.error('Error: Cannot checkResetCode no UserContext.Provider element in parents');
  },
  email: null,
  firstname: null,
  lastname: null,
  username: null,
  userAccess: false,
  admin: false,
};

export const UserContext = React.createContext(Context);

export interface SignupParams {
  email: string;
  username: string;
  firstname: string;
  lastname: string;
  password: string;
};

export interface UpdateUserParams {
  email?: string;
  username?: string;
  firstname?: string;
  lastname?: string;
}

export interface IUserProps {
  initialState?: IUserState;
}

export interface IUserState {
  email: string;
  firstname: string;
  lastname: string;
  username: string;
  userAccess: boolean;
  admin: boolean;
}

export interface UserWrapperMethods {
  signup: (params: SignupParams, callback: (error) => void) => void;
  getUser: () => void;
  login: (username: string, password: string, callback: (error) => void) => void;
  logout: () => void;
  updateUser: (params: UpdateUserParams, callback: (error) => void) => void;
  updatePassword: (oldpassword: string, newpassword: string, callback: (error) => void) => void;
  forgotPassword: (email: string, callback: (error) => void) => void;
  resetPassword: (newpassword: string, userId: string, reset_code: string, callback: (error) => void) => void;
  checkResetCode: (userId: string, reset_code: string, callback: (error) => void) => void;
}

export class UserWrapper extends React.Component<IUserProps, IUserState> implements UserWrapperMethods {
  public static defaultProps: IUserProps = {
    initialState: {
      email: null,
      firstname: null,
      lastname: null,
      username: null,
      userAccess: false,
      admin: false,
    },
  }
  constructor(props) {
    super(props)
    this.state = this.props.initialState;
    this.signup = this.signup.bind(this);
    this.getUser = this.getUser.bind(this);
    this.login = this.login.bind(this);
    this.logout = this.logout.bind(this);
    this.updateUser = this.updateUser.bind(this);
    this.updatePassword = this.updatePassword.bind(this);
    this.forgotPassword = this.forgotPassword.bind(this);
  }

  getUser() {
    const request = new Request('/api/user/', {
      method: 'GET',
      credentials: 'same-origin',
      headers: {
          'Content-Type': 'application/json',
          'Accept': 'application/json',
      },
    });
    fetch(request)
    .then((response) => {
      if (response.status == 200) {
        return response.json();
      } else {
        const error = new Error(response.statusText)
        error.message = JSON.stringify(response)
        throw error;
      }
    })
    .then((body: IUserContext) => {
      let user: IUserState = {
        email: body.email,
        username: body.username,
        userAccess: true,
        firstname: body.firstname,
        lastname: body.lastname,
        admin: body.admin,
      };
      this.setState(() => {
        return user;
      });
    })
    .catch((error) => {
      console.log(error);
    });
  }

  signup(params, callback) {

    const payload: SignupParams = {
      email: params.email,
      username: params.username,
      firstname: params.firstname,
      lastname: params.lastname,
      password: params.password,
    }
    
    const request = new Request('/api/user/', {
      method: 'POST',
      credentials: 'same-origin',
      headers: {
          'Content-Type': 'application/json',
          'Accept': 'application/json',
      },
      body: JSON.stringify(payload),   
    });

    fetch(request)
    .then((response) => {
      return response.json();
    })
    .then((body: IUserContext) => {
      let user: IUserState = {
        email: body.email,
        username: body.username,
        userAccess: true,
        firstname: body.firstname,
        lastname: body.lastname,
        admin: body.admin,
      };
      this.setState(() => {
        return user;
      });
      callback(false);
    })
    .catch((error) => {
      callback(error);
      console.error(error);
    });
  }

  login(email, password, callback) {
    const payload = {
      email: email,
      password: password,
    }
    
    const request = new Request('/api/user/login', {
      method: 'POST',
      credentials: 'same-origin',
      headers: {
          'Content-Type': 'application/json',
          'Accept': 'application/json',
      },
      body: JSON.stringify(payload),   
    });

    fetch(request)
    .then((response) => {
      if (response.status == 200) {
        return response.json();
      } else {
        const error = new Error(response.statusText)
        error.message = JSON.stringify(response)
        throw error;
      }
    })
    .then((body: IUserContext) => {
      let user: IUserState = {
        email: body.email,
        username: body.username,
        userAccess: true,
        firstname: body.firstname,
        lastname: body.lastname,
        admin: body.admin,
      };
      this.setState(() => {
        return user;
      });
      callback(false);
    })
    .catch((error) => {
      callback(error);
      console.error(error);
    });
  }

  logout() {
    const request = new Request('/api/user/logout', {
      method: 'GET',
      credentials: 'same-origin',
      headers: {
          'Content-Type': 'application/json',
          'Accept': 'application/json',
      },
    });

    fetch(request)
    .then((response) => {
      if (response.status == 200) {
        let user: IUserState = {
          email: null,
          username: null,
          userAccess: false,
          firstname: null,
          lastname: null,
          admin: false,
        };
        this.setState(() => {
          return user;
        });
      } else {
        console.log('logout failed');
      }
    })
    .catch((error) => {
      console.error(error);
    });
  }

  updateUser(params, callback) {
    let payload:UpdateUserParams = {};
    if ('email' in params) {payload.email = params.email}
    if ('username' in params) {payload.username = params.username}
    if ('firstname' in params) {payload.firstname = params.firstname}
    if ('lastname' in params) {payload.lastname = params.lastname}

    if (Object.keys(payload).length > 0) {
      const request = new Request('/api/user/', {
        method: 'PUT',
        credentials: 'same-origin',
        headers: {
          'Content-Type': 'application/json',
          'Accept': 'application/json',
        },
        body: JSON.stringify(payload),
      });

      fetch(request)
      .then((response) => {
        return response.json();
      })
      .then((body: IUserContext) => {  
        let user: IUserState = {
          email: body.email,
          username: body.username,
          userAccess: true,
          firstname: body.firstname,
          lastname: body.lastname,
          admin: body.admin,
        };

        this.setState(() => {
          return user;
        });
        callback(false);
      })
      .catch((error) => {
        callback(true);
        console.log(error);
      });
    }

  }

  updatePassword(oldpassword, newpassword, callback) {
    const payload = {
      oldpassword: oldpassword,
      newpassword: newpassword
    };
    if (Object.keys(payload).length > 0) {
      const request = new Request('/api/user/login', {
        method: 'PUT',
        credentials: 'same-origin',
        headers: {
          'Content-Type': 'application/json',
          'Accept': 'application/json',
        },
        body: JSON.stringify(payload),
      });

      fetch(request)
      .then((response) => {
        return response.json();
      })
      .then((body: IUserContext) => {  
        let user: IUserState = {
          email: body.email,
          username: body.username,
          userAccess: true,
          firstname: body.firstname,
          lastname: body.lastname,
          admin: body.admin,
        };

        this.setState(() => {
          return user;
        });
        callback(false);
      })
      .catch((error) => {
        callback(true);
        console.log(error);
      });
    }
  }
  
  forgotPassword(email, callback) {
    const payload = {
      email: email,
    };
    if (Object.keys(payload).length > 0) {
      const request = new Request('/api/user/forgotpassword', {
        method: 'POST',
        credentials: 'same-origin',
        headers: {
          'Content-Type': 'application/json',
          'Accept': 'application/json',
        },
        body: JSON.stringify(payload),
      });

      fetch(request)
      .then((response) => {
        return response.json();
      })
      .then((body) => {
        callback(true);
      })
      .catch((error) => {
        callback(false);
        console.log(error);
      });
    }
  }  
  
  resetPassword(newpassword, userId, reset_code, callback) {
    const payload = {
      newpassword: newpassword,
    };
    if (Object.keys(payload).length > 0) {
      const request = new Request(`/api/user/resetpassword/${userId}/${reset_code}`, {
        method: 'POST',
        credentials: 'same-origin',
        headers: {
          'Content-Type': 'application/json',
          'Accept': 'application/json',
        },
        body: JSON.stringify(payload),
      });

      fetch(request)
      .then((response) => {
        return response.json();
      })
      .then((body:{
        vaild: boolean
      }) => {
        callback(!body.vaild);
      })
      .catch((error) => {
        callback(true);
        console.log(error);
      });
    }
  }

  checkResetCode(userId, reset_code, callback) {
    const request = new Request(`/api/user/resetpassword/${userId}/${reset_code}`, {
      method: 'GET',
      credentials: 'same-origin',
      headers: {
        'Content-Type': 'application/json',
        'Accept': 'application/json',
      },
    });

    fetch(request)
    .then((response) => {
      console.log(response)
      return response.json();
    })
    .then((body:{
      vaild: boolean
    }) => {
      callback(body.vaild);
    })
    .catch((error) => {
      callback(false);
      console.log(error);
    });
  }

  render() {
    const {signup, getUser, login, logout, updateUser, updatePassword, forgotPassword, resetPassword, checkResetCode } = this;
    const { email, firstname, lastname, username, userAccess, admin } = this.state;
      const context: IUserContext = {
        email: email,
        firstname: firstname,
        lastname: lastname,
        username: username,
        userAccess: userAccess,
        admin: admin,
        signup: signup,
        getUser: getUser,
        login: login,
        logout: logout,
        updateUser: updateUser,
        updatePassword: updatePassword,
        forgotPassword: forgotPassword,
        resetPassword: resetPassword,
        checkResetCode: checkResetCode,
      }
      return (
          <UserContext.Provider value={context} >
              {this.props.children}
          </UserContext.Provider>
      );
  }
}


export default UserWrapper;

export interface userProviderOptions {
  props?: IUserProps;
}

export const userProvider = (Component, options?: userProviderOptions) => {
  class UserProvider extends React.Component<null, null> {
    render () {
      let userWrapperProps: IUserProps = {};
      if (options) {
        if ('props' in options) {
          userWrapperProps = options.props;
        }
      }
      return (
        <UserWrapper {...userWrapperProps}>
          <Component {...this.props} />  
        </UserWrapper>
      );
    }
  }
  
    return UserProvider;
}

export const userConsumer = (Component) => {
    class UserConsumer extends React.Component<any> {
    render () {
        return (
          <UserContext.Consumer>
            {(context) => {
  
              return (
                <Component {...this.props} {...{user: context}} />
              )
            }}
          </UserContext.Consumer>
        );
      }
    }
  
    return UserConsumer;
}