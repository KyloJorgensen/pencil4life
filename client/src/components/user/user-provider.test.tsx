'use strict';

import * as React from 'react';
import { shallow, mount, ShallowWrapper, ReactWrapper } from 'enzyme';
import * as fetch from 'jest-fetch-mock';
import { UserContext, userProvider, userConsumer, UserWrapper, Context, UpdateUserParams } from './user-provider';

let wrapper: ShallowWrapper | ReactWrapper;

class Elem extends React.Component {
	render() {
		return (
			<p></p>
		)
	}
}

const methodTest = (methodTest) => {
    interface MethodTestProps {
        user: Context;
    }
    class MethodTest extends React.Component<MethodTestProps> {
        constructor(props) {
            super(props);
        }
        componentDidMount() {
            methodTest(this.props.user);
        }
        render() {
            const { email, firstname, lastname, username, userAccess, admin } = this.props.user;
            return (
                <div>
                    <p className="email">{email}</p> 
                    <p className="username">{username}</p>   
                    <p className="firstname">{firstname}</p>    
                    <p className="lastname">{lastname}</p>    
                    <p className="userAccess">{userAccess ? 'true' : 'false' }</p>       
                    <p className="admin">{admin ? 'true' : 'false' }</p>    
                </div>
            );
        }
    }
    return MethodTest;
}
describe('User-Provider', () => {
    beforeAll(() => {
        fetch.resetMocks()
    })

    describe('Render UserConext.Consumer', () => {
        it('should match the snaphot', () => {
            wrapper = mount(
                <UserContext.Consumer>
                    {context => {
                        return (
                            <p>a</p>
                        );
                    }}
                </UserContext.Consumer>
            );
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('Render UserWrapper', () => {
        it('matches the snapshot', () => {
            wrapper = shallow(<UserWrapper />);
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('Render userProvider', () => {
        it('matches the snapshot', () => {
            const UP = userProvider(Elem);
            wrapper = shallow(<UP />);
            expect(wrapper).toMatchSnapshot();
        });
        
        it('should contain User Wrapper ', () => {
            const UP = userProvider(Elem);
            wrapper = mount(
                <UP />
            );

            const conatins = wrapper.contains(
                <UserWrapper>
                    <Elem />
                </UserWrapper>
            );

            expect(conatins).toEqual(true);
        });
    });

    describe('Render userConsumer', () => {
        it('matches the snapshot', () => {
            const UC = userConsumer(Elem);
            wrapper = shallow(<UC />);
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('signup', () => {
        const UP = userProvider(userConsumer(methodTest((user: Context) => {user.signup({
            email: 'test@example.com',
            username: 'coolguy123',
            firstname: 'Joe',
            lastname: 'Doe',
            password: 'password',
        })})));

        const body = JSON.stringify({
            email: 'test@example.com',
            username: 'coolguy123',
            firstname: 'Joe',
            lastname: 'Doe',
            admin: false,
        });

        beforeEach(() => {
            fetch.resetMocks()
            fetch.mockResponses(
                // [ JSON.stringify({}), { status: 200 }],
                [ body, { status: 200 }],
            );
            wrapper = mount(<UP />);
        });

        it('should give a email', () => {
            expect(wrapper.find('p.email').text()).toEqual('test@example.com');
        });
        
        it('should give a username', () => {
            expect(wrapper.find('p.username').text()).toEqual('coolguy123');
        });
        
        it('should give a firstname', () => {
            expect(wrapper.find('p.firstname').text()).toEqual('Joe');
        });
        
        it('should give a lastname', () => {
            expect(wrapper.find('p.lastname').text()).toEqual('Doe');
        });
        
        it('should give a userAccess', () => {
            expect(wrapper.find('p.userAccess').text()).toEqual('true');
        });
        
        it('should give a admin', () => {
            expect(wrapper.find('p.admin').text()).toEqual('false');
        });
    });

    describe('getUser', () => {
        const UP = userProvider(userConsumer(methodTest((user: Context) => {user.getUser()})));

        const body = JSON.stringify({
            email: 'test@example.com',
            username: 'coolguy123',
            firstname: 'Joe',
            lastname: 'Doe',
            admin: false,
        });

        beforeEach(() => {
            fetch.resetMocks()
            fetch.once(body);
            wrapper = mount(<UP />);
        });

        it('should give a email', () => {
            expect(wrapper.find('p.email').text()).toEqual('test@example.com');
        });
        
        it('should give a username', () => {
            expect(wrapper.find('p.username').text()).toEqual('coolguy123');
        });
        
        it('should give a firstname', () => {
            expect(wrapper.find('p.firstname').text()).toEqual('Joe');
        });
        
        it('should give a lastname', () => {
            expect(wrapper.find('p.lastname').text()).toEqual('Doe');
        });
        
        it('should give a userAccess', () => {
            expect(wrapper.find('p.userAccess').text()).toEqual('true');
        });
        
        it('should give a admin', () => {
            expect(wrapper.find('p.admin').text()).toEqual('false');
        });
    });

    describe('login', () => {
        const UP = userProvider(userConsumer(methodTest((user: Context) => {user.login('coolguy', 'password')})));

        const body = JSON.stringify({
            email: 'test@example.com',
            username: 'coolguy123',
            firstname: 'Joe',
            lastname: 'Doe',
            admin: false,
        });

        beforeEach(() => {
            fetch.resetMocks()
            fetch.mockResponses(
                // [ JSON.stringify({}), { status: 200 }],
                [ body, { status: 200 }],
            );
            wrapper = mount(<UP />);
        });

        it('should give a email', () => {
            expect(wrapper.find('p.email').text()).toEqual('test@example.com');
        });
        
        it('should give a username', () => {
            expect(wrapper.find('p.username').text()).toEqual('coolguy123');
        });
        
        it('should give a firstname', () => {
            expect(wrapper.find('p.firstname').text()).toEqual('Joe');
        });
        
        it('should give a lastname', () => {
            expect(wrapper.find('p.lastname').text()).toEqual('Doe');
        });
        
        it('should give a userAccess', () => {
            expect(wrapper.find('p.userAccess').text()).toEqual('true');
        });
        
        it('should give a admin', () => {
            expect(wrapper.find('p.admin').text()).toEqual('false');
        });
    });

    describe('logout', () => {
        const UP = userProvider(userConsumer(methodTest((user: Context) => {user.logout()})),{
            props: {
                initialState: {
                    email: 'test@example.com',
                    username: 'coolguy123',
                    firstname: 'Joe',
                    lastname: 'Doe',
                    userAccess: true,
                    admin: false,
                },
            },
        });

        const body = JSON.stringify({
        });

        beforeEach(() => {
            fetch.resetMocks()
            fetch.mockResponses(
                [ JSON.stringify({}), { status: 200 }],
                // [ body, { status: 200 }],
            );
            wrapper = mount(<UP />);
        });

        it('should give a email', () => {
            expect(wrapper.find('p.email').text()).toEqual('');
        });
        
        it('should give a username', () => {
            expect(wrapper.find('p.username').text()).toEqual('');
        });
        
        it('should give a firstname', () => {
            expect(wrapper.find('p.firstname').text()).toEqual('');
        });
        
        it('should give a lastname', () => {
            expect(wrapper.find('p.lastname').text()).toEqual('');
        });
        
        it('should give a userAccess', () => {
            expect(wrapper.find('p.userAccess').text()).toEqual('false');
        });
        
        it('should give a admin', () => {
            expect(wrapper.find('p.admin').text()).toEqual('false');
        });
    });

    describe('updateUser', () => {
        const params: UpdateUserParams = {
            email: 'test2@gamil.com',
            username: 'coolestguy123',
            firstname: 'Joesph',
            lastname: 'Smith',
        }
        const UP = userProvider(userConsumer(methodTest((user: Context) => {user.updateUser(params)})));


        const body = JSON.stringify({
            email: 'test2@example.com',
            username: 'coolestguy123',
            firstname: 'Joesph',
            lastname: 'Smith',
            admin: false,
        });

        beforeEach(() => {
            fetch.resetMocks()
            fetch.mockResponses(
                [ body, { status: 200 }],
            );
            wrapper = mount(<UP />);
        });

        it('should give a email', () => {
            expect(wrapper.find('p.email').text()).toEqual('test2@example.com');
        });
        
        it('should give a username', () => {
            expect(wrapper.find('p.username').text()).toEqual('coolestguy123');
        });
        
        it('should give a firstname', () => {
            expect(wrapper.find('p.firstname').text()).toEqual('Joesph');
        });
        
        it('should give a lastname', () => {
            expect(wrapper.find('p.lastname').text()).toEqual('Smith');
        });
        
        it('should give a userAccess', () => {
            expect(wrapper.find('p.userAccess').text()).toEqual('true');
        });
        
        it('should give a admin', () => {
            expect(wrapper.find('p.admin').text()).toEqual('false');
        });
    });
});
