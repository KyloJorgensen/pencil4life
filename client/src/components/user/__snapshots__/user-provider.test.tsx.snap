// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User-Provider Render UserConext.Consumer should match the snaphot 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <Context.Consumer>
    [Function]
  </Context.Consumer>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": <p>
      a
    </p>,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "a",
    },
    "ref": null,
    "rendered": Array [
      "a",
    ],
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": <p>
        a
      </p>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "a",
      },
      "ref": null,
      "rendered": Array [
        "a",
      ],
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`User-Provider Render UserWrapper matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserWrapper
    initialState={
      Object {
        "admin": false,
        "email": null,
        "firstname": null,
        "lastname": null,
        "userAccess": false,
        "username": null,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": undefined,
      "value": Object {
        "admin": false,
        "checkResetCode": [Function],
        "email": null,
        "firstname": null,
        "forgotPassword": [Function],
        "getUser": [Function],
        "lastname": null,
        "login": [Function],
        "logout": [Function],
        "resetPassword": [Function],
        "signup": [Function],
        "updatePassword": [Function],
        "updateUser": [Function],
        "userAccess": false,
        "username": null,
      },
    },
    "ref": null,
    "rendered": null,
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_changedBits": 0,
        "_changedBits2": 0,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": Object {
          "admin": false,
          "checkResetCode": [Function],
          "email": null,
          "firstname": null,
          "forgotPassword": [Function],
          "getUser": [Function],
          "lastname": null,
          "login": [Function],
          "logout": [Function],
          "resetPassword": [Function],
          "signup": [Function],
          "updatePassword": [Function],
          "updateUser": [Function],
          "userAccess": false,
          "username": null,
        },
        "_currentValue2": Object {
          "admin": false,
          "checkResetCode": [Function],
          "email": null,
          "firstname": null,
          "forgotPassword": [Function],
          "getUser": [Function],
          "lastname": null,
          "login": [Function],
          "logout": [Function],
          "resetPassword": [Function],
          "signup": [Function],
          "updatePassword": [Function],
          "updateUser": [Function],
          "userAccess": false,
          "username": null,
        },
        "_defaultValue": Object {
          "admin": false,
          "checkResetCode": [Function],
          "email": null,
          "firstname": null,
          "forgotPassword": [Function],
          "getUser": [Function],
          "lastname": null,
          "login": [Function],
          "logout": [Function],
          "resetPassword": [Function],
          "signup": [Function],
          "updatePassword": [Function],
          "updateUser": [Function],
          "userAccess": false,
          "username": null,
        },
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": undefined,
        "value": Object {
          "admin": false,
          "checkResetCode": [Function],
          "email": null,
          "firstname": null,
          "forgotPassword": [Function],
          "getUser": [Function],
          "lastname": null,
          "login": [Function],
          "logout": [Function],
          "resetPassword": [Function],
          "signup": [Function],
          "updatePassword": [Function],
          "updateUser": [Function],
          "userAccess": false,
          "username": null,
        },
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_changedBits": 0,
          "_changedBits2": 0,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": Object {
            "admin": false,
            "checkResetCode": [Function],
            "email": null,
            "firstname": null,
            "forgotPassword": [Function],
            "getUser": [Function],
            "lastname": null,
            "login": [Function],
            "logout": [Function],
            "resetPassword": [Function],
            "signup": [Function],
            "updatePassword": [Function],
            "updateUser": [Function],
            "userAccess": false,
            "username": null,
          },
          "_currentValue2": Object {
            "admin": false,
            "checkResetCode": [Function],
            "email": null,
            "firstname": null,
            "forgotPassword": [Function],
            "getUser": [Function],
            "lastname": null,
            "login": [Function],
            "logout": [Function],
            "resetPassword": [Function],
            "signup": [Function],
            "updatePassword": [Function],
            "updateUser": [Function],
            "userAccess": false,
            "username": null,
          },
          "_defaultValue": Object {
            "admin": false,
            "checkResetCode": [Function],
            "email": null,
            "firstname": null,
            "forgotPassword": [Function],
            "getUser": [Function],
            "lastname": null,
            "login": [Function],
            "logout": [Function],
            "resetPassword": [Function],
            "signup": [Function],
            "updatePassword": [Function],
            "updateUser": [Function],
            "userAccess": false,
            "username": null,
          },
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`User-Provider Render userConsumer matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserConsumer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "Consumer": [Circular],
      "Provider": Object {
        "$$typeof": Symbol(react.provider),
        "_context": [Circular],
      },
      "_calculateChangedBits": null,
      "_changedBits": 0,
      "_changedBits2": 0,
      "_currentRenderer": Object {},
      "_currentRenderer2": null,
      "_currentValue": Object {
        "admin": false,
        "checkResetCode": [Function],
        "email": null,
        "firstname": null,
        "forgotPassword": [Function],
        "getUser": [Function],
        "lastname": null,
        "login": [Function],
        "logout": [Function],
        "resetPassword": [Function],
        "signup": [Function],
        "updatePassword": [Function],
        "updateUser": [Function],
        "userAccess": false,
        "username": null,
      },
      "_currentValue2": Object {
        "admin": false,
        "checkResetCode": [Function],
        "email": null,
        "firstname": null,
        "forgotPassword": [Function],
        "getUser": [Function],
        "lastname": null,
        "login": [Function],
        "logout": [Function],
        "resetPassword": [Function],
        "signup": [Function],
        "updatePassword": [Function],
        "updateUser": [Function],
        "userAccess": false,
        "username": null,
      },
      "_defaultValue": Object {
        "admin": false,
        "checkResetCode": [Function],
        "email": null,
        "firstname": null,
        "forgotPassword": [Function],
        "getUser": [Function],
        "lastname": null,
        "login": [Function],
        "logout": [Function],
        "resetPassword": [Function],
        "signup": [Function],
        "updatePassword": [Function],
        "updateUser": [Function],
        "userAccess": false,
        "username": null,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_changedBits": 0,
        "_changedBits2": 0,
        "_currentRenderer": Object {},
        "_currentRenderer2": null,
        "_currentValue": Object {
          "admin": false,
          "checkResetCode": [Function],
          "email": null,
          "firstname": null,
          "forgotPassword": [Function],
          "getUser": [Function],
          "lastname": null,
          "login": [Function],
          "logout": [Function],
          "resetPassword": [Function],
          "signup": [Function],
          "updatePassword": [Function],
          "updateUser": [Function],
          "userAccess": false,
          "username": null,
        },
        "_currentValue2": Object {
          "admin": false,
          "checkResetCode": [Function],
          "email": null,
          "firstname": null,
          "forgotPassword": [Function],
          "getUser": [Function],
          "lastname": null,
          "login": [Function],
          "logout": [Function],
          "resetPassword": [Function],
          "signup": [Function],
          "updatePassword": [Function],
          "updateUser": [Function],
          "userAccess": false,
          "username": null,
        },
        "_defaultValue": Object {
          "admin": false,
          "checkResetCode": [Function],
          "email": null,
          "firstname": null,
          "forgotPassword": [Function],
          "getUser": [Function],
          "lastname": null,
          "login": [Function],
          "logout": [Function],
          "resetPassword": [Function],
          "signup": [Function],
          "updatePassword": [Function],
          "updateUser": [Function],
          "userAccess": false,
          "username": null,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`User-Provider Render userProvider matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserProvider />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Elem />,
      "initialState": Object {
        "admin": false,
        "email": null,
        "firstname": null,
        "lastname": null,
        "userAccess": false,
        "username": null,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Elem />,
        "initialState": Object {
          "admin": false,
          "email": null,
          "firstname": null,
          "lastname": null,
          "userAccess": false,
          "username": null,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
